src/loader.rs

```diff
@@ -83,10 +83,36 @@ impl Chunk {
     }
 }
 
-#[derive(Debug)]
+/// Workaround for the fact that rust-s3 doesn't redact credentials in debug
+/// impls.
+#[allow(clippy::ptr_arg)]
```

Comment:
why do you want to explicitly take a `&Vec` instead of a slice? `format_with` doesn't have the same requirements as the actual debug trait.